controllers/nifiparametercontext_controller.go

```diff
@@ -223,9 +224,21 @@ func (r *NifiParameterContextReconciler) Reconcile(ctx context.Context, req ctrl
 		r.Recorder.Event(instance, corev1.EventTypeNormal, "Creating",
 			fmt.Sprintf("Creating parameter context %s", instance.Name))
 
-		status, err := parametercontext.CreateParameterContext(instance, parameterSecrets, clientConfig)
-		if err != nil {
-			return RequeueWithError(r.Log, "failure creating parameter context", err)
+		var status *v1alpha1.NifiParameterContextStatus
+
+		if instance.Spec.IsTakeOverEnabled() {
```

Comment:
Assume that we have an existing parameter context in a nifi with name `my-pm` and that we've configured a parameter context in nifikop with the same name.

Suppose takeover is not enabled for that parameter context w/ nifikop. Then we don't look up the pre-existing parameter context and attempt to create the parameter context because `status` will be `nil`. That will result in a failure since nifi disallows creating parameter contexts with the same name, right?

If taking over existing parameter contexts is truly the intent of this change, then i think that if takeOver is not enabled and there's a pre-existing parameter context with the same name then the reconciliation loop should end right there. That is, nifikop should not modify pre-existing parameter contexts unless takeOver is enabled.

Reply:
Yes, NiFi forbids creating parameter contexts with the same name.
As for why I did it like, It's simply because I have copied the way it was done for the NiFi User (https://github.com/konpyutaika/nifikop/blob/master/controllers/nifiuser_controller.go#L286).

I wanted to keep the current behavior but maybe you are right I should not keep this behavior and stop the reconcile if the parameter context already exists and the takeOver is disable.

I will update my PR .

Reply:
Ah i see. In that case i think it can be corrected like so:

```go
status, err = parametercontext.FindParameterContextByName(instance, clientConfig)
if err != nil {
  return RequeueWithError(r.Log, "failure finding parameter context", err)
}
if status == nil {
  // Create NiFi parameter context
  status, err = parametercontext.CreateParameterContext(instance, parameterSecrets, clientConfig)
  if err != nil {
    return RequeueWithError(r.Log, "failure creating parameter context", err)
  }
} else {
 // the parameter context already exists
 if !instance.Spec.IsTakeOverEnabled() {
   return RequeueWithError(r.Log, "Parameter context X already exists and takeOver is not enabled, so not syncing parameter context", err)
 }
}
```

or something like that :)